### PR TITLE
Pass '-U' option if available to command 'info' and 'check' of qemu-img

### PR DESCRIPTION
**Pull request by morecache**
qemu commit 244a566 introduces image locking to perm operations.
When we use qemu 2.10 or newer, run some qemu testcases with qcow2 image,
If the host OFD lock is available and vt.setup.keep_guest_running is True
or option 'check_image' is yes, The qemu-img info or the qemu-img check
reports error due to it can't obtain the shared "write" lock on opened
images. This patch add '-U'(force shared permissions) to fixes the issue.

Signed-off-by: Lin Ma <lma@suse.com>

### CODE REVIEW

**Comment by morecache** [comment]
# avocado run --vt-type qemu --vt-extra-params check_image=yes --vt-guest-os Linux.OpenSUSE.42.3.x86_64.i440fx type_specific.io-github-autotest-qemu.block_stream.simple_test.start_stream
......
 (1/1) type_specific.io-github-autotest-qemu.block_stream.simple_test.start_stream: FAIL (88.83 s)
......

part of log:
2017-08-08 15:49:42,356 stacktrace       L0042 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado_plugins_vt-51.0-py2.7.egg/avocado_vt/test.py:245
2017-08-08 15:49:42,503 stacktrace       L0045 ERROR| Traceback (most recent call last):
2017-08-08 15:49:42,503 stacktrace       L0045 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-51.0-py2.7.egg/avocado_vt/test.py", line 398, in _runTest
2017-08-08 15:49:42,503 stacktrace       L0045 ERROR|     run_func(self, params, env)
2017-08-08 15:49:42,503 stacktrace       L0045 ERROR|   File "/usr/lib/python2.7/site-packages/autotest/client/shared/error.py", line 140, in new_fn
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|     return fn(*args, **kwargs)
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_stream.py", line 38, in run
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|     image_size = stream_test.get_image_size(image_file)
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|   File "/var/lib/avocado/data/avocado-vt/test-providers.d/downloads/io-github-autotest-qemu/qemu/tests/block_stream.py", line 16, in get_image_size
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|     info = utils.system_output(cmd)
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|   File "/usr/lib/python2.7/site-packages/autotest/client/shared/utils.py", line 1270, in system_output
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|     verbose=verbose, args=args).stdout
2017-08-08 15:49:42,504 stacktrace       L0045 ERROR|   File "/usr/lib/python2.7/site-packages/autotest/client/shared/utils.py", line 918, in run
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR|     "Command returned non-zero exit status")
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR| CmdError: Command </usr/bin/qemu-img info /var/lib/avocado/data/avocado-vt/images/openSUSE-Leap-42.3-64.qcow2> failed, rc=1, Command returned non-zero exit status    [context: (run)]
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR| * Command:
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR|     /usr/bin/qemu-img info /var/lib/avocado/data/avocado-vt/images/openSUSE-
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR|     Leap-42.3-64.qcow2
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR| Exit status: 1
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR| Duration: 0.0965659618378
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR|
2017-08-08 15:49:42,505 stacktrace       L0045 ERROR| stderr:
2017-08-08 15:49:42,506 stacktrace       L0045 ERROR| qemu-img: Could not open '/var/lib/avocado/data/avocado-vt/images/openSUSE-Leap-42.3-64.qcow2': Failed to get shared "write" lock
2017-08-08 15:49:42,506 stacktrace       L0045 ERROR| Is another process using the image?
2017-08-08 15:49:42,506 stacktrace       L0046 ERROR|
2017-08-08 15:49:42,506 stacktrace       L0039 ERROR|
2017-08-08 15:49:42,506 stacktrace       L0042 ERROR| Reproduced traceback from: /usr/lib/python2.7/site-packages/avocado/core/test.py:596
2017-08-08 15:49:42,507 stacktrace       L0045 ERROR| Traceback (most recent call last):
2017-08-08 15:49:42,507 stacktrace       L0045 ERROR|   File "/usr/lib/python2.7/site-packages/avocado_plugins_vt-51.0-py2.7.egg/avocado_vt/test.py", line 272, in runTest
2017-08-08 15:49:42,507 stacktrace       L0045 ERROR|     raise self.__status
2017-08-08 15:49:42,507 stacktrace       L0045 ERROR| CmdError: Command </usr/bin/qemu-img info /var/lib/avocado/data/avocado-vt/images/openSUSE-Leap-42.3-64.qcow2> failed, rc=1, Command returned non-zero exit status    [context: (run)]
2017-08-08 15:49:42,507 stacktrace       L0045 ERROR| * Command:
2017-08-08 15:49:42,507 stacktrace       L0045 ERROR|     /usr/bin/qemu-img info /var/lib/avocado/data/avocado-vt/images/openSUSE-
2017-08-08 15:49:42,508 stacktrace       L0045 ERROR|     Leap-42.3-64.qcow2
2017-08-08 15:49:42,508 stacktrace       L0045 ERROR| Exit status: 1
2017-08-08 15:49:42,508 stacktrace       L0045 ERROR| Duration: 0.0965659618378
2017-08-08 15:49:42,508 stacktrace       L0045 ERROR|
2017-08-08 15:49:42,508 stacktrace       L0045 ERROR| stderr:
2017-08-08 15:49:42,508 stacktrace       L0045 ERROR| qemu-img: Could not open '/var/lib/avocado/data/avocado-vt/images/openSUSE-Leap-42.3-64.qcow2': Failed to get shared "write" lock
2017-08-08 15:49:42,508 stacktrace       L0045 ERROR| Is another process using the image?

**Comment by luckyh** [comment]
Well, I think this implementation is just a workaround for the issue, many other places would call `qemu-img` as well, so it needs to be fixed at the API level.

Anyway, I'll merge this fix if we do need it for passing the current tests.

File: qemu/tests/block_stream.py [hunk 1]
```diff
@@ -12,7 +12,11 @@ class BlockStreamTest(blk_stream.BlockStream):
 
     def get_image_size(self, image_file):
         qemu_img = utils_misc.get_qemu_img_binary(self.params)
-        cmd = "%s info %s" % (qemu_img, image_file)
+        # '-U' was added into help output by qemu commit a8d16f9
+        if '-U' in utils.system_output("%s --help" % qemu_img):
+            cmd = "%s info -U %s" % (qemu_img, image_file)
+        else:
+            cmd = "%s info %s" % (qemu_img, image_file)
```
Comment (vivianQizhu) [inline qemu/tests/block_stream.py]: This part has already been fixed by https://github.com/autotest/tp-qemu/pull/1145

**Comment by vivianQizhu** [comment]
This pull request should be out of date, the `-U` issue has already been fixed by https://github.com/avocado-framework/avocado-vt/pull/1228
So let's close this one. Thanks.

**Pull request closed without merging**